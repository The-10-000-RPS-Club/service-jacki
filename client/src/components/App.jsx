import React, { useState, useEffect } from 'react';

import axios from 'axios';

import styled from 'styled-components';

import QuestionList from './QuestionList.jsx';

import DropDown from './DropDown.jsx';

import LoadMore from './LoadMore.jsx';



function App() {

	const [questions, setQuestions] = useState([]);
	const [numQuestions, setNumQuestions] = useState(5);
	// const [sort, setSort] = useState(null);

	const getQuestions = () => {
		axios.get('/api/products/questions')
			.then((data) => setQuestions(questions.concat(data.data)))
			.catch((err) => (err));
	};
	
	const incrementHelpfulCount = (quesId, ansId, option) => {
		axios.patch(`/api/products/questions/${quesId}/${ansId}/${option}`)
			.then((data) => console.log(`${option} count increased`))
			.catch((err) => (err));
	};

	useEffect(() => {
    getQuestions();
	}, [numQuestions]);
	

console.log(numQuestions, questions.length);
  return (
    <div>
      <div>
        <Wrapper>
          <div>
            <QuestionButton>Ask a question</QuestionButton>
            <Title>
							<div>Questions & Answers</div>
							</Title>
           <DropDown />
          </div>
        </Wrapper>
        <div>
          <QuestionList questions={questions} incrementHelpfulCount={incrementHelpfulCount}/>
        </div>
      </div>
      <div>
      <LoadMore numQuestions={numQuestions} setNumQuestions={setNumQuestions} getQuestions={getQuestions}/>
			</div>
    </div>
  );
}

const Wrapper = styled.section`

margin: auto;
width: 55%;
position: relative;
padding: 10px;
font-family: Stuart, Georgia, serif;
`;

const Title = styled.div`
font-family: Stuart, Georgia, serif;
font-size: 22px;
padding 10px;
position: relative;
padding-bottom: 30px;
padding-left: 0px;
`;

const QuestionButton = styled.button`
background-color: #006699;
color: white;
float: right;
border: none;
padding: 10px;
border-radius: 2px;
font-family: "Roboto","Helvetica Neue","Helvetica","Arial",sans-serif;
font-size: 16px;
`;

export default App;
