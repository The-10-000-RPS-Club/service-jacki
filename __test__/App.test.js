import React from 'react';
import Enzyme, { shallow, mount, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import App from '../client/src/components/App.jsx';

Enzyme.configure({ adapter: new Adapter() });

describe('App component tests', () => {
  it('should render without throwing an error', () => {
    expect(shallow(<App />).contains(<div id="list-container">Bar</div>)).toBe(false);
  });
});
