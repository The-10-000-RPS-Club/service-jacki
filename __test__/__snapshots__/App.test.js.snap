// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A suite example using Snapshot renders App component correctly 1`] = `
<div>
  <div>
    <section
      className="sc-fnVYJo dPttRp"
    >
      <div>
        <button
          className="sc-fFSRdu CtOHj"
        >
          Ask a question
        </button>
        <h3>
          Questions & Answers
        </h3>
        <div
          className="sc-bkbjAj ifuNii"
        >
          <p>
            Sort by: 
            <select>
              <option
                href="#"
              >
                Newest questions
              </option>
              <option
                href="#"
              >
                Newest answers
              </option>
              <option
                href="#"
              >
                Most answered
              </option>
              <option
                href="#"
              >
                Answers Needed
              </option>
              <option
                href="#"
              >
                Most helpful answers
              </option>
            </select>
          </p>
        </div>
      </div>
    </section>
    <div>
      <div>
        <section
          className="sc-hBMVcZ chhbmF"
        >
          <p>
            0
             Questions
          </p>
        </section>
      </div>
    </div>
  </div>
  <div>
    <button
      className="sc-iemXMA egDyyf"
      onClick={[Function]}
    >
      Load more
    </button>
  </div>
</div>
`;

exports[`A suite example using Snapshot renders App correctly 1`] = `
<div>
  <div>
    <section
      className="sc-fnVYJo dPttRp"
    >
      <div>
        <button
          className="sc-fFSRdu CtOHj"
        >
          Ask a question
        </button>
        <h3>
          Questions & Answers
        </h3>
        <div
          className="sc-bkbjAj ifuNii"
        >
          <p>
            Sort by: 
            <select>
              <option
                href="#"
              >
                Newest questions
              </option>
              <option
                href="#"
              >
                Newest answers
              </option>
              <option
                href="#"
              >
                Most answered
              </option>
              <option
                href="#"
              >
                Answers Needed
              </option>
              <option
                href="#"
              >
                Most helpful answers
              </option>
            </select>
          </p>
        </div>
      </div>
    </section>
    <div>
      <div>
        <section
          className="sc-hBMVcZ chhbmF"
        >
          <p>
            0
             Questions
          </p>
        </section>
      </div>
    </div>
  </div>
  <div>
    <button
      className="sc-iemXMA egDyyf"
      onClick={[Function]}
    >
      Load more
    </button>
  </div>
</div>
`;
